From 3087fdb45ddc1ad70763faa6115633ffc3cd6bea Mon Sep 17 00:00:00 2001
From: Jianxiong Pan <jianxiong.pan@amlogic.com>
Date: Wed, 9 Mar 2022 12:46:32 +0800
Subject: [PATCH 26/95] mm: add mem debug support on arm 32bit. [2/2]

PD#SWPL-72254

Problem:
no mem debug.

Solution:
porting from kernel5.4.

Verify:
local.

Change-Id: I03597ffe7cec4fc473bfd5a9c67bae09cba50b61
Signed-off-by: Jianxiong Pan <jianxiong.pan@amlogic.com>
---
 arch/arm/mm/init.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mm/init.c b/arch/arm/mm/init.c
index 6162a070a410..6d04cb75d26b 100644
--- a/arch/arm/mm/init.c
+++ b/arch/arm/mm/init.c
@@ -41,6 +41,10 @@
 
 #include "mm.h"
 
+#ifdef CONFIG_AMLOGIC_MEM_DEBUG
+#include <linux/amlogic/mem_debug.h>
+#endif
+
 #ifdef CONFIG_CPU_CP15_MMU
 unsigned long __init __clear_cr(unsigned long mask)
 {
@@ -331,6 +335,9 @@ void __init mem_init(void)
 
 	free_highpages();
 
+#ifdef CONFIG_AMLOGIC_MEM_DEBUG
+	dump_mem_layout_boot_phase();
+#endif
 	/*
 	 * Check boundaries twice: Some fundamental inconsistencies can
 	 * be detected at build time already.
-- 
2.25.1

